src/main/java/org/rumbledb/expressions/comparison/ComparisonExpression.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors: Stefan Irimescu, Can Berker Cikis
+ *
+ */
+
+package org.rumbledb.expressions.comparison;
+
+import org.rumbledb.exceptions.ExceptionMetadata;
+import org.rumbledb.exceptions.OurBadException;
+import org.rumbledb.expressions.AbstractNodeVisitor;
+import org.rumbledb.expressions.Expression;
+import org.rumbledb.expressions.Node;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ComparisonExpression extends Expression {
+
+    public static enum ComparisonOperator {
+        VC_EQ("eq"),
+        VC_NE("ne"),
+        VC_LT("lt"),
+        VC_LE("le"),
+        VC_GT("gt"),
+        VC_GE("ge"),
+        GC_EQ("="),
+        GC_NE("!="),
+        GC_LT("<"),
+        GC_LE("<="),
+        GC_GT(">"),
+        GC_GE(">=");
+
+        private String name;
+
+        ComparisonOperator(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public String toString() {
+            return this.name;
+        }
+
+        public boolean isValueComparison() {
+            return this == VC_EQ
+                ||
+                this == VC_NE
+                ||
+                this == VC_LT
+                ||
+                this == VC_LE
+                ||
+                this == VC_GT
+                ||
+                this == VC_GE;
+        }
```

Comment:
How about introducing isValueComparison as a property on the enums themselves ? Just like a name. I think It would take up more memory, but would reduce the needed computation with equality checks.